src/Common/EventSyntaxExtensions.cs

```diff
@@ -5,6 +5,30 @@ namespace Moq.Analyzers.Common;
 /// </summary>
 internal static class EventSyntaxExtensions
 {
+    /// <summary>
+    /// Extracts the event name from a lambda selector of the form: x => x.EventName += null.
+    /// </summary>
+    /// <param name="semanticModel">The semantic model.</param>
+    /// <param name="eventSelector">The lambda event selector expression.</param>
+    /// <param name="eventName">The extracted event name, if found.</param>
+    /// <returns><see langword="true" /> if the event name was found; otherwise, <see langword="false" />.</returns>
+    internal static bool TryGetEventNameFromLambdaSelector(
```

Comment:
As an extension of `SemanticModel`, this should move to `./src/Common/SemanticModelExtensions.cs`